keyboards/xd87/config.h

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2018 Alexander Fougner <fougner89 at gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x0000
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    KP Republic
+#define PRODUCT         XD87 TKL
+#define DESCRIPTION     KP Republic XD87 ATmega 32u4
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 17
+
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+*/
+
+
+#define MATRIX_ROW_PINS { D1, B0, B1, C7, D3, D5 }
+
+#define MATRIX_COL_PINS { E6, F0, F1, F4, F5, F6, F7, B5, B6, C6, D4, D6, D7, B4, B2, B3, D2 }
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL, or CUSTOM_MATRIX */
+#define DIODE_DIRECTION COL2ROW
+
+// #define BACKLIGHT_PIN B7
```

Comment:
This pcb actually supports both underglow(need to solder it yourself) and backlighting.

Reply:
That is correct, I wasn't considering that because I don't use it myself, but I can see if I find the correct pin for it.

Reply:
it's the D0 pin, however it doesn't seem to work very well for me by just defining the BACKLIGHT_PIN. The backlight is always on, and I can not toggle it. However, I think I can see very tiny differences when increasing/decreasing the levels.

The pin is connected to the gate of a P-channel MOSFET, which in turn drives all the LEDs. Anyone got any ideas?